libs/injection/bbhnet/injection/injection.py

```diff
@@ -0,0 +1,285 @@
+    snr = np.zeros(n_sample)
+
+    ifo = bilby.gw.detector.get_empty_interferometer(ifo)
+    b, a = sig.butter(
+        N=8,
+        Wn=waveform_generator.waveform_arguments["minimum_frequency"],
+        btype="highpass",
+        fs=waveform_generator.sampling_frequency,
+    )
+    for i, p in enumerate(sample_params):
+
+        # For less ugly function calls later on
+        ra = p["ra"]
+        dec = p["dec"]
+        geocent_time = p["geocent_time"]
+        psi = p["psi"]
+
+        # Generate signal in IFO
+        polarizations = waveform_generator.time_domain_strain(p)
+        signal = np.zeros(waveform_size)
+        for mode, polarization in polarizations.items():
+            # Get ifo response
+            response = ifo.antenna_response(ra, dec, geocent_time, psi, mode)
+            signal += response * sig.filtfilt(b, a, polarization)
+
+        # Total shift = shift to trigger time + geometric shift
+        dt = waveform_duration / 2.0
+        dt += ifo.time_delay_from_geocenter(ra, dec, geocent_time)
+        signal = np.roll(signal, int(np.round(dt * sample_rate)))
+
+        # Calculate SNR
+        if noise_psd is not None:
+            if get_snr:
+                snr[i] = calc_snr(signal, noise_psd, sample_rate)
+
+        signals[i] = signal
+    if get_snr:
+        return signals, snr
+    return signals
+
+
+def inject_signals(
+    frame_files: Iterable[str],
+    channels: [str],
+    ifos: [str],
+    prior_file: str,
+    n_samples: int,
+    outdir: str,
+    fmin: float = 20,
+    waveform_duration: float = 8,
+    snr_range: Iterable[float] = [25, 50],
+):
+
+    """Injects simulated BBH signals into a frame, or set of corresponding
+    frames from different interferometers. Frames should have the same
+    start/stop time and the same sample rate
+
+    Arguments:
+    - frame_files: list of paths to frames to be injected
+    - channels: channel names of the strain data in each frame
+    - ifos: list of interferometers corresponding to frames, e.g., H1, L1
+    - prior_file: prior file for bilby to sample from
+    - n_samples: number of signal to inject
+    - outdir: output directory to which injected frames will be written
+    - fmin: Minimum frequency for highpass filter
+    - waveform_duration: length of injected waveforms
+    - snr_range: desired signal SNR range
+    """
+
+    strains = [
+        TimeSeries.read(frame, ch) for frame, ch in zip(frame_files, channels)
+    ]
+
+    logging.info("Read strain from frame files")
+
+    span = set([strain.span for strain in strains])
+    if len(span) != 1:
+        print("Error: The given strains have different durations")
+        exit()
+    frame_start, frame_stop = next(iter(span))
+    frame_duration = frame_stop - frame_start
+
+    sample_rate = set([int(strain.sample_rate.value) for strain in strains])
+    if len(sample_rate) != 1:
+        print("Error: The given strains have different sample rates")
+        exit()
```

Comment:
Raise errors here, e.g. `raise ValueError("Frame files {} and {} have different durations".format(*frame_files))`